homeworks/week6/trang_vu6.java

```diff
@@ -0,0 +1,65 @@
+ 
+ class trang_vu6{
+  public static void main(String[] args) {
+    System.out.println(getShortestSubstringOfAllCharacters("xyyzyzyx", "xyz"));  // "zyx"
+  }
+
+  // Time complexity O(n)
```

Comment:
O(N) assuming `pat.length is a constant`.

To be generic, we can just say O(N+M) where N and M are length of two input strings accordingly.